packages/rich-text/src/plugins/Table/createTablePlugin.ts

```diff
@@ -66,10 +72,12 @@ export const createTablePlugin = (): RichTextPlugin =>
       editor.insertFragment = (fragments) => {
         // We need to make sure we have a new, empty and clean paragraph in order to paste tables as-is due to how Slate behaves
         // More info: https://github.com/ianstormtaylor/slate/pull/4489 and https://github.com/ianstormtaylor/slate/issues/4542
-        const fragmentHasTable = fragments.some(
-          (fragment) => (fragment as CustomElement).type === BLOCKS.TABLE
-        );
-        if (fragmentHasTable) {
+        const isInsertingTable = fragments.some((fragment) => isTable(fragment as CustomElement));
+        const isTableFirstFragment =
+          fragments.findIndex((fragment) => isTable(fragment as CustomElement)) === 0;
+        const currentLineHasText = getText(editor, editor.selection?.focus.path) !== '';
+
+        if (isInsertingTable && isTableFirstFragment && currentLineHasText) {
```

Comment:
It can't be `isTableFirstFragment: true` if `isInsertingTable: false` right?
```suggestion
        const isTableFirstFragment =
          fragments.findIndex((fragment) => isTable(fragment as CustomElement)) === 0;
        const currentLineHasText = getText(editor, editor.selection?.focus.path) !== '';

        if (isTableFirstFragment && currentLineHasText) {
```

Reply:
It can't be. The other way around is true, I can insert table, but it might not be the first element on the clipboard.

`table` = `isTableFirstFragment: true` and `isInsertingTable: true`
`table + p` = `isTableFirstFragment: true` and `isInsertingTable: true`
`p + ul + table` = `isTableFirstFragment: false` and `isInsertingTable: true`
`p` = `isTableFirstFragment: false` and `isInsertingTable: false`